src/components/common/CertifySection.vue

```diff
@@ -52,6 +55,12 @@ export default class CertifySection extends Mixins(DateMixin, SharedMixin) {
   /** Whether to perform validation. */
   @Prop({ default: false }) readonly validate: boolean
 
+  /** To determine whether user input is enabled. */
+  @Prop({ required: true }) readonly disableEdit!: boolean
+
+  /** To determine if user is staff. */
+  @Prop({ required: true }) readonly isStaff!: boolean
```

Comment:
You're "accepting" this prop but you're not passing it in from any of the views that use this component.

Option 1: You could pass it in in all 3 views. Pro: this component doesn't need to know about the store. Con: a bit more code.

Option 2: You could just declare the store getter for it here. Pro/con: the opposite of option 1.

On the one hand, I think it's OK if this component imports the store. On the other hand, the 3 views already import isRoleStaff so it would be easy to just pass the prop in.

Another consideration is: does this component need to know that "is staff" means it's editable? That's pretty specific. Maybe this component should just be told whether it's editable, and the caller should determine the logic.

Reply:
So for option 1 I should pass in a value in each of the three views ie) `:isStaff="isRoleStaff"` and have each view insert `@Getter isRoleStaff boolean` in each of the views and this would sync the isStaff prop with the store?

For option 2 I would just declare the store getter ie) `@Getter isRoleStaff boolean`

Reply:
I think you understand it :)